crates/sui-types/src/zk_login_authenticator.rs

```diff
@@ -164,19 +172,26 @@ impl AuthenticatorTrait for ZkLoginAuthenticator {
             SignatureScheme::ZkLoginAuthenticator,
         )?;
 
-        // Use flag || pk_bytes.
-        let mut extended_pk_bytes = vec![self.user_signature.scheme().flag()];
-        extended_pk_bytes.extend(self.user_signature.public_key_bytes());
-        verify_zk_login(
-            &self.inputs,
-            self.max_epoch,
-            &extended_pk_bytes,
-            &aux_verify_data.oidc_provider_jwks,
-            &aux_verify_data.zk_login_env,
-        )
-        .map_err(|e| SuiError::InvalidSignature {
-            error: e.to_string(),
-        })
+        if zklogin_inputs_cache.is_some()
+            && zklogin_inputs_cache.unwrap().is_cached(&self.hash_inputs())
```

Comment:
what ensures that the proof is for the pk verified above?

Reply:
its cached only when its verified. hash_inputs() calculated based on the zklogin inputs and pk bytes and max_epoch